projects/controllable_dialogue/README.md

```diff
@@ -24,11 +26,254 @@ improvements in human quality judgments.
+- `word2count.pkl`: This is a pickle file containing information necessary to compute NIDF measures, needed for the specificity control methods
+
+Place both `ConvAI2_parlaiformat` and `ConvAI2_controllable` in your `ParlAI/data` directory.
+
+## (Alternatively) Make the data yourself
+
+Here are the commands to create ConvAI2_parlaiformat and ConvAI2_controllable yourself:
+
+To create `ConvAI2_parlaiformat`:
+
+    cd ParlAI
+
+    mkdir data/ConvAI2_parlaiformat
+
+    python parlai/scripts/convert_data_to_parlai_format.py \
+    -t convai2:SelfOriginal:no_cands -dt train:ordered \
+    -of data/ConvAI2_parlaiformat/train.txt
+
+    python parlai/scripts/convert_data_to_parlai_format.py \
+    -t convai2:SelfOriginal:no_cands -dt valid \
+    -of data/ConvAI2_parlaiformat/valid.txt
+
+To create `ConvAI2_controllable`:
+
+First, make a directory:
+
+    cd ParlAI
+    mkdir data/ConvAI2_controllable
+
+Next, create `word2count.pkl`:
+
+    python projects/controllable_dialogue/controllable_seq2seq/nidf.py
+
+This will create a file called `word2count.pkl` in your `ConvAI2_controllable` directory. It might take a while, especially the part when it goes through the Twitter dataset counting words. Once it's done, go to `nidf.py` and enter the filepath to your `word2count.pkl` file.
+
+Next, create `arora.pkl`:
+
+    python projects/controllable_dialogue/controllable_seq2seq/arora.py
+
+This will create a file called `arora.pkl` in your `ConvAI2_controllable` directory. It might take a while -- in particular, it will download GloVe vectors and store them in `ParlAI/data/ConvAI2_controllable/glove_vectors`, which usually takes some time. Once it's done, go to `arora.py` and enter the filepath to your `arora.pkl` file and your `glove_vectors` directory.
+
+Next, create `ConvAI2_controllable/train.txt` and `valid.txt`:
+
+    python projects/controllable_dialogue/make_control_dataset.py \
+      --fromfile_datapath data/ConvAI2_parlaiformat/train.txt \
+      --outfile data/ConvAI2_controllable/train.txt \
+      --controls question,lastuttsim,avg_nidf
+
+    python $CD/make_control_dataset.py \
+      --fromfile_datapath $PA/data/ConvAI2_parlaiformat/valid.txt \
+      --outfile $PA/data/ConvAI2_controllable/valid.txt \
+      --controls question,lastuttsim,avg_nidf
+
+This will create files called `train.txt` and `valid.txt` in your `ConvAI2_controllable` directory.
+
+Note: Due to changes in ParlAI, there might be some small differences between the generated files obtained in this section, and the downloaded files in the previous section.
+
+## Download the pretrained models
+
+Download the models and place them in `ParlAI/data/models/controllable_dialogue`.
+
+TODO: update these instructions for downloading via ParlAI.
+
+The directory `controllable_dialogue` contains the following models:
+
+- `twitter_pretrained_baseline`: a seq2seq model trained on the Twitter dataset. TODO: is there a reference for where else this model can be downloaded?
```

Comment:
it *might* be this one?

https://github.com/facebookresearch/ParlAI/blob/52364b2466124561138a23fc53fd3061118ac1b4/projects/twitter/seq2seq/eval_f1.py#L14-L22